src/components/PrivateRoute/PrivateRoute.jsx

```diff
@@ -1,11 +1,16 @@
-import { Redirect, Route } from 'react-router-dom';
+import { Redirect, Route, useLocation } from 'react-router-dom';
 import { useUser } from '../../context/UserContext';
 
 // A wrapper for <Route> that redirects to the login
 // screen if you're not yet authenticated.
 export default function PrivateRoute({ children, ...rest }) {
   // TODO: Use the user in context to determine whether to redirect to /login
-  // const { user } = useUser();
+  const { user } = useUser();
+  const location = useLocation();
 
-  return <Route {...rest} render={({ location }) => children} />;
+  if(user?.id) {
+
+    return <Route {...rest}>{children}</Route>;
+  }
+  return <Redirect to={{ pathname: '/login', state: { from: location }}} />
```

Comment:
You have an extra return here. You want to return the `<Route >` component with a conditional render prop that either displays the `{children}` or the `<Redirect>` component based on whether the user is present in context or not.